lua/nvim-treesitter/indent.lua

```diff
@@ -17,12 +17,19 @@ local function getline(lnum)
   return vim.api.nvim_buf_get_lines(0, lnum - 1, lnum, false)[1] or ""
 end
 
+---@param lnum integer
+---@return integer
+local function get_indentcols_at_line(lnum)
+  local _, indentcols = getline(lnum):find "^%s*"
+  return indentcols or 0
+end
```

Comment:
For a line of
`\t\t\t\tfoo`

It should return tabstop * 4, but instead it's returning tabstop.

Reply:
Wouldn't that cause the exact same issue as before? In (new) lines 160-178, the number of indentation columns is used as the position argument to get a treesitter node via `get_first_node_at_line` and `get_last_node_at_line`, which both call `TSNode:descendant_for_range`. That function expects a column number, not a "visual" column number. For example consider
```python
def foo():
				x = 1 # indent using 4 tabs
```
For this example, `root:get_descendent_for_range(1, 4):type()` (i.e., 2nd line 5th column, corresponding to the character `x`) correctly returns `identifier`. On the other hand `root:get_descendent_for_range(1, 16):type()` returns `module`. 

The point is that `indentcols` is the number of actual columns, not visual indentation, and this behaviour is required to get the correct treesitter node. The function `get_indentcols_at_line` is only being wherever a column position is required to get a treesitter node, not to get the actual indent width.

Reply:
Makes perfect sense honestly. This is correct